ql/indexes/indexmanager.cpp

```diff
@@ -45,14 +58,23 @@ namespace QuantLib {
         return temp;
     }
 
-    void IndexManager::clearHistory(const std::string& name) { data_.erase(name); }
+    void IndexManager::clearHistory(const std::string& name) {
+        notifier(name)->notifyObservers();
+        data_.erase(name);
+        notifiers_.erase(name);
```

Comment:
Hmm, no, the notifier should stay because we have existing index instances registered with it that won't re-register with the new one when it's recreated.  Maybe this hints at a a missing test?

Reply:
You are right. The notifiers stay, but we call notifyObservers() when we clear a history. Yes?

Reply:
Correct.